hw_diag/diagnostics/add_gateway_txn_diagnostic.py

```diff
@@ -0,0 +1,40 @@
+import secrets
+
+from hm_pyhelper.diagnostics import Diagnostic, DiagnosticsReport
+from hm_pyhelper.constants.shipping import DESTINATION_WALLETS_KEY, DESTINATION_ADD_GATEWAY_TXN_KEY
+from hm_pyhelper.miner_json_rpc.exceptions import MinerMalformedAddGatewayTxn
+
+from hw_diag.utilities.miner import create_add_gateway_txn
+from hw_diag.utilities.security import get_verified_json
+
+
+class AddGatewayTxnDiagnostic(Diagnostic):
+    # Error messages
+    NO_DESTINATION_WALLETS_MSG = "Destination wallets not found in payload JSON."
+    VERIFICATION_FAILED_MSG = "Verifying the payload signature failed."
+
+    def __init__(self, shipping_destination_with_signature):
```

Comment:
missing type hint

Reply:
Now, all the methods of the all diagnostics are type hinted.